dm/dm/master/server.go

```diff
@@ -1502,14 +1503,21 @@ func (s *Server) generateSubTask(
 	task string,
 	cliArgs *config.TaskCliArgs,
 ) (*config.TaskConfig, []*config.SubTaskConfig, error) {
+	var err error
 	cfg := config.NewTaskConfig()
 	// bypass the meta check by set any value. If start-time is specified, DM-worker will not use meta field.
 	if cliArgs != nil && cliArgs.StartTime != "" {
+		err = cfg.RawDecode(task)
+		if err != nil {
+			return nil, nil, terror.WithClass(err, terror.ClassDMMaster)
+		}
 		for _, inst := range cfg.MySQLInstances {
-			inst.Meta = &config.Meta{BinLogName: cliArgs.StartTime}
+			inst.Meta = &config.Meta{BinLogName: binlog.FakeBinlogName}
 		}
+		err = cfg.Adjust()
+	} else {
+		err = cfg.Decode(task)
 	}
```

Comment:
Can we set inst.Meta after cfg.Decode? So we no need RawDecode and Adjust, only depend on cfg.Decode.

Reply:
`cfg.Decode` will call `adjust` internally, then it will report error about inst.Meta.